datasetinsights/commands/download.py

```diff
@@ -80,51 +73,19 @@ def convert(self, value, param, ctx):
     ),
 )
 @click.option(
-    "--dataset-version",
-    type=click.STRING,
-    default=const.DEFAULT_DATASET_VERSION,
-    help=(
-        "The dataset version. This only applies to some dataset "
-        "where different versions are available."
-    ),
-)
-@click.option(
-    "--project-id",
-    type=click.STRING,
+    "--access-token",
+    type=str,
     default=None,
-    help=(
-        "Unity project-id used for Unity Simulation runs. This will override "
-        "synthetic datasets source-uri for Unity Simulation."
-    ),
-)
-@click.option(
-    "--run-execution-id",
-    type=click.STRING,
-    default=None,
-    help=(
-        "Unity Simulation run-execution-id. This will override synthetic "
-        "datasets source-uri for Unity Simulation."
-    ),
-)
-@click.option(
-    "--auth-token",
-    type=click.STRING,
-    default=None,
-    help=(
-        "Unity Simulation auth token. This will override synthetic "
-        "datasets source-uri for Unity Simulation."
-    ),
+    help="Unity Simulation access token. "
+    "This will override synthetic datasets source-uri for Unity Simulation",
 )
 def cli(
-    name,
-    data_root,
-    source_uri,
-    include_binary,
-    dataset_version,
-    project_id,
-    run_execution_id,
-    auth_token,
+    source_uri, output, include_binary, access_token,
 ):
     ctx = click.get_current_context()
     logger.debug(f"Called download command with parameters: {ctx.params}")
-    # TODO: Call download dataset command here.
+
+    downloader = DownloaderFactory.find(source_uri)(access_token=access_token)
```

Comment:
The name interface is a little confusing here. `DownloaderFactory.find` will actually create a downloader instance, which suggests this method should be renamed. What about a `create_downloader` method in `downloader/base.py`, and a separate method `_find_downloader`, as we will have similar pattern for `Estimators`, `Dataset`, `EvaluationMetric`,...?

Reply:
DownloaderFactory.find does not create an instance of the downloader it only returns the class. the construction happens in this line. (access_token=access_token)

Reply:
I'll convert it into a function instead of a static method